koordinates/gui/login_widget.py

```diff
@@ -247,9 +286,18 @@ def _auth_finished(self, key):
 
         try:
             KoordinatesClient.instance().login(key)
-            self.store_api_key()
+            self.store_api_key(key, refresh_token)
             self.login_button.set_state(AuthState.LoggedIn)
             self.open_login_window_label.hide()
+
+            self.oauth_refresh_timer = QTimer(self)
+            self.oauth_refresh_timer.setSingleShot(True)
+            # request refresh after 9 hours, even though the token
+            # expires after 10...
+            self.oauth_refresh_timer.setInterval(32400*1000)
+            self.oauth_refresh_timer.timeout.connect(self._refresh_auth)
+            self.oauth_refresh_timer.start()
```

Comment:
How does QTimer behave across sleep/resume? ie: obviously our API tokens expire at a specific time rather than a relative time that pauses/resumes with system sleep.

Reply:
The timers aren't paused during suspend. So if the system is woken before the timeout would have otherwise occurred then all is good. If the system was still asleep when the timeout should have happened then it will trigger on resume. In this case it will try to refresh immediately, fail, and trigger the full auth workflow again.